### DISCUSSION TITLE
Surface Mode

### DISCUSSION THREAD

**Discussion by gratefulfrog** [supermerill/SuperSlicer#1301] [Q&A]
Hello,

I am trying to print RC plane profiles on a Voron 2.4.

The requirement is currently defined in terms of Cura's "Surface mode" cf. This page [PlanePrint printing](https://www.planeprint.com/print)

Could anyone tell me if an equivalent setup is possible with Superslicer?  I know that Ideamaker, also a prusaslicer fork, now offers Surface mode in the current beta release, but I am not sure how to set up a Voron 2.4 profile in ideamaker.

I realize that this question is really far out there, and I am sorry to be such a noobie...

Any help or suggestions would be great!

Thanks,
Bob

**Comment by supermerill** [comment]
* set top layers to 0
* set bottom layers to 0
* set infill percentage to 0%
* 1 perimeter
* thin walls activated, maybe.

**Comment by gratefulfrog** [comment]
Hi!
Thanks for your reply. But it does not work as desired.  Internal structure is lost from the model...

model view
![image](https://user-images.githubusercontent.com/1917335/121858702-2bdfae80-ccf7-11eb-8894-ed1440b6762e.png)

sliced:
![Screenshot_sliced](https://user-images.githubusercontent.com/1917335/121859068-8f69dc00-ccf7-11eb-88b0-dd62fe3ad071.png)

I am sorry to pose such noobie questions...
Cheers,
Bob
model file:
[PLANEPRINT_WINGTEST 1.20.zip](https://github.com/supermerill/SuperSlicer/files/6646567/PLANEPRINT_WINGTEST.1.20.zip)

**Reply by supermerill** [reply]
triangle normal are completely off in your model.
Imo, this stl is a hack designed for a version of cura, and that's it.
I won't support that kind of thing.
If someone wants a feature for a need, i can implement it, but I won't support stl hacks designed for other software.

**Comment by gratefulfrog** [comment]
Hi
I'm not trying to push you to do something, sorry if that is what it seems.  My goal is to print RC planes on a Voron2.4 and it is not easy going for me since I am not an expert.  

I have been experimenting with ideamaker because at my local fablab (Graulhet France) they have a Raise3D printer.  Raise sent me a beta of ideamaker that implements surface mode that seems to work. And since Ideamaker is also a fork of prusalsicer/slice3r, I was hoping there was a way to use SuperSlicer since that seems to be the official "slicer of preference" of the Voron community.

I hope I have not offended you - It was not at all my intension nor desire. I like SuperSlicer!!! I was just hoping...

Cheers,
Bob

**Reply by supermerill** [reply]
np.
It's just that that stl is a cura hack.

**Comment by gratefulfrog** [comment]
Sorry to insist, but exactly do you mean by a hack? 

I believe that the stl was made in Blender (but I could be wrong) and it displays ok in Meshlab as well as Superslicer and Ideamaker.  

It slices as desired in both Cura and I think in Ideamaker. 

I am just trying to understand for my own use and in discussions with Radio Control hobbyist for 3D printing planes on a Voron.

Thanks for your help!

**Reply by supermerill** [reply]
the top & bottom surface has their normal upside-down, so you can see through it, but to have a solid volume, it needs to be in the opposite direction.
Then, inside the volume, you have some triangles than you describe as 'Internal structure'. I don't know how it is supposed to be sliced: it's already inside the piece!Maybe cura process all that in a very specific/weird way...

**Comment by gratefulfrog** [comment]
sorry, it was designed in Rhino, not Blender...

**Comment by neophyl** [comment]
This has been a long running issue with Prusa Slicer that has gotten quite acrimonious at times.  If you would care to read the end of this closed issue from PS https://github.com/prusa3d/PrusaSlicer/issues/3062  you will see that 2.4 which has yet to be released has added in a new optional slicing mode that is *supposed* to handle these non standard non-manifold models.

These RC models are NOT valid geometry strictly speaking and only certain slicers can slice them usually as they take advantage or certain tricks.  That's why its called a hack.  From the explanation in the PS issue the new optional mode will make PS slice using one of these other techniques and so also make use of the same way.

I'm guessing that Merill will understand the specifics more than me :)

**Reply by gratefulfrog** [reply]
@neophyl  Thank you for this detailed explanation and the reference to the long issue-thread which I read with great interest, yet also great sadness to see the angry reactions of some, many participants.  Quite frankly, I was shocked.  I have been a software developer for nearly all my life, and a computer scientist for my adult life; I have developed in many languages, environments, situations, both for work and for pleasure and in the FOSS world.  I have a lot of experience of dealing with users, being the primary interface to users, as a software architect and team manager and so on..

The discussion about what is a valid model as input to the slicer is really not the issue, in my opinion. When one makes something and puts it out there, one cannot imagine how others may want to use it. That is what the the Hacking/Making world is all about - _re-purposing things for novel usage._  When I was 12 years old, I made a hang-glider from bamboo poles that were used to spool carpets and a painter's tarp, held together with packing tape. No one expected those things to be combined into a hang-glider, but they were; Yes it was a hack, but no one got angry about it !

The use of a "mode switch" is standard software engineering for dealing with nonstandard situations. Cura has "surface mode" ; Ideamaker has "Single Shell Surface Mode".  I was merely suggesting that SuperSlicer could have such a switch as well. 

Of course, when I wrote the original suggestion I did not know about all this history, and anger. Had I known, I would never have made this "issue".

I am very so sorry to have started this discussion and now withdraw my naive request. I will simply try to use other tools for the job. 

My only goal was to help a nice project move forward and make it easier to use of my future Voron  2.4 printer the way it seems that most Vorons are used, i.e. with gcode generated by SuperSlicer.

I had no idea that I was throwing oil on a blazing fire and regret it immensely.

Please, @neophyl  @supermerill , and any others who I may have offended, accept my apologies.

**Reply by neophyl** [reply]
The reason I linked that was to give some background but also that as it is probably ‘fixed’ in 2.4 then it’s likely that at some point in the future that the same option will also make its way into Susi too eventually.

**Reply by gratefulfrog** [reply]
Thank you. I see. I hope so. I remain saddened by this experience.

**Reply by supermerill** [reply]
Sorry if my post was perceived as "angry", that was not the purpose.
I just wanted to be as concise as possible.
Maybe it's better with a smiley? ;-p

**Reply by gratefulfrog** [reply]
Thansk for the update - it's all ok!  I am not easily offended, I was more surprised by the other thread on the prusslicer issue... 

It's all good, life goes on, and in the end it's only rock-n-roll... :-D

Sorry to have bothered y'all 
Ciao,
Bob

**Reply by noradtux** [reply]
Chiming in here because after having to deal with Cura to print this kind of stuff which I didn't quite enjoy I'd like to try to explain what Cura does in "Surface Mode": As I understand it, in SM Cura completely drops the assumption of an object being a volume. It just traces the lines of the polygons. These models are made to make use of this mode, so they have their "infill" built in as polygons to make it as light as possible while being as strong as required. And no, I don't expect this to be implemented. I just want to clarify :)

**Reply by supermerill** [reply]
there will be a special mode in 2.4 for that kind of stl.

**Comment by KubiProjects** [comment]
Hi,

I have the same issue now and ran into this thread. 
@supermerill, you have mentioned that there will be a special mode in 2.4. Have you managed to implement something or dropped the idea as it is an ugly hack?
In any way, thanks a lot for the great tool!

-- EDIT --
I tried Even/Odd slicing mode, but no success :(

https://www.planeprint.com/print

-- EDIT 2 --
Same issue with https://cults3d.com/en/3d-model/various/rc-plane-su-29-1100mm wing parts.

I'm trying to contact the author to understand how he made the internal structures that are missing. I'm not an expert on STLs to find out if it is the same problem as with the planeprint models.

Anyway, I'm trying to design my model in freecad and do not want to make the same mistake (?)...

**Reply by mszulen** [reply]
Yeah, Even/Odd is not the same as Surface mode in Cura. This has been already tested/discussed by the community.
From Ultimaker's website: 
![image](https://github.com/supermerill/SuperSlicer/assets/6932464/bbe6e788-68bb-4d9b-8421-3f4d96d890d9)
